core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/heuristic/selector/entity/EntitySelectorFactory.java

```diff
@@ -186,9 +186,10 @@ private EntitySelector<Solution_> applyNearbySelection(HeuristicConfigPolicy<Sol
                 entitySelectorFactory.buildEntitySelector(configPolicy, minimumCacheType, resolvedSelectionOrder);
         NearbyDistanceMeter nearbyDistanceMeter = ConfigUtils.newInstance(nearbySelectionConfig, "nearbyDistanceMeterClass",
                 nearbySelectionConfig.getNearbyDistanceMeterClass());
+        NearbyDistanceMeter wrappedNearbyDistanceMeter = new WrappedNearbyDistanceMeter<>(nearbyDistanceMeter);
```

Comment:
This wrapped class, and other wrapped classes here, exist so that I don't need to propagate equality down to all of these interfaces, such as selection sorters, nearby meters etc. Instead, the wrapper implements the equality check; if the origin class is the same, the instance is equal.

Since we create the instances, these instances are totally interchangeable, and therefore the equality check based on class makes sense.

Reply:
Would it be possible to avoid the wrapping by only have 1 instance of the same nearbyDistanceMeterClass?

In the grander scheme of things, it should be possible to declare the NearbyDistanceMeter as a method annotation once, so all nearby selectors just grab it. Having to configure it by selector separately is a pain and error prone.

Following that, the nearby distance meter - and more importantly - the nearby distance matrix should become a supply (if it isn't already) that can be reused by multiple selectors. Things get more complex when we factor in scoped supplies (not yet implemented) that have certain lifecycle, but we don't want to kill the CH distance matrix just the rebuild it from scratch for LS. Building a distance matrix for 10 000 locations is _very_ expensive.